src/compiler/parser.ts

```diff
@@ -3859,13 +3859,14 @@ module ts {
         function parseObjectBindingElement(): BindingElement {
             let node = <BindingElement>createNode(SyntaxKind.BindingElement);
             // TODO(andersh): Handle computed properties
-            let id = parsePropertyName();
-            if (id.kind === SyntaxKind.Identifier && token !== SyntaxKind.ColonToken) {
-                node.name = <Identifier>id;
+            let tokenIsIdentifier = isIdentifier();

```

Comment:
Did we decide to not just always parse out a property name, and then error later if the property name turns out to be a keyword?